File: CC_C#/BVCS Projects/Sandwich/Form1.cs
#region Using directives

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Windows.Forms;

#endregion

namespace Sandwich
{
    partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        int breadChoice;
        int cheeseChoice;
        int numberMeats;
        int numberCondiments;

        private void Form1_Load(object sender, EventArgs e)
        {
            // Initialize bread and cheese choices
            breadChoice = 1;
            cheeseChoice = 0;
        }

        private void rdoWhite_CheckedChanged(object sender, EventArgs e)
        {
            // White bread selected
            breadChoice = 1;
        }

        private void rdoWheat_CheckedChanged(object sender, EventArgs e)
        {
            // Wheat bread selected
            breadChoice = 2;
        }

        private void rdoRye_CheckedChanged(object sender, EventArgs e)
        {
            // Rye bread selected
            breadChoice = 3;
        }

        private void rdoNone_CheckedChanged(object sender, EventArgs e)
        {
            // No cheese selected
            cheeseChoice = 0;
        }

        private void rdoAmerican_CheckedChanged(object sender, EventArgs e)
        {
            // American cheese selected
            cheeseChoice = 1;
        }

        private void rdoSwiss_CheckedChanged(object sender, EventArgs e)
        {
            // Swiss cheese selected
            cheeseChoice = 2;
        }

        private void btnOrder_Click(object sender, EventArgs e)
        {
            // Start Text with bread type
            txtOrder.Text = "Sandwich Order:\r\n";
            switch (breadChoice)
            {
                case 1:
                    txtOrder.Text = txtOrder.Text + "White Bread\r\n";
                    break;
                case 2:
                    txtOrder.Text = txtOrder.Text + "Wheat Bread\r\n";
                    break;
                case 3:
                    txtOrder.Text = txtOrder.Text + "Rye Bread\r\n";
                    break;
            }
            // Add and count meats
            numberMeats = 0;
            if (chkRoastBeef.Checked)
            {
                numberMeats = numberMeats + 1;
                txtOrder.Text = txtOrder.Text + "Roast Beef\r\n";
            }
            if (chkHam.Checked)
            {
                numberMeats = numberMeats + 1;
                txtOrder.Text = txtOrder.Text + "Ham\r\n";
            }
            if (chkTurkey.Checked)
            {
                numberMeats = numberMeats + 1;
                txtOrder.Text = txtOrder.Text + "Turkey\r\n";
            }
            if (chkPastrami.Checked)
            {
                numberMeats = numberMeats + 1;
                txtOrder.Text = txtOrder.Text + "Pastrami\r\n";
            }
            if (chkSalami.Checked)
            {
                numberMeats = numberMeats + 1;
                txtOrder.Text = txtOrder.Text + "Salami\r\n";
            }
            // If no meats picked, say so
            if (numberMeats == 0)
            {
                txtOrder.Text = txtOrder.Text + "No Meat\r\n";
            }
            // Add cheese type
            switch (cheeseChoice)
            {
                case 0:
                    txtOrder.Text = txtOrder.Text + "No Cheese\r\n";
                    break;
                case 1:
                    txtOrder.Text = txtOrder.Text + "American Cheese\r\n";
                    break;
                case 2:
                    txtOrder.Text = txtOrder.Text + "Swiss Cheese\r\n";
                    break;
            }
            // Finally, add  and count condiments
            numberCondiments = 0;
            if (chkMustard.Checked)
            {
                numberCondiments=numberCondiments+1;
                txtOrder.Text = txtOrder.Text + "Mustard\r\n";
            }
            if (chkMayo.Checked)
            {
                numberCondiments=numberCondiments+1;
                txtOrder.Text = txtOrder.Text + "Mayonnaise\r\n";
            }
            if (chkLettuce.Checked)
            {
                numberCondiments=numberCondiments+1;
                txtOrder.Text = txtOrder.Text + "Lettuce\r\n";
            }
            if (chkTomato.Checked)
            {
                numberCondiments=numberCondiments+1;
                txtOrder.Text = txtOrder.Text + "Tomato\r\n";
            }
            if (chkOnions.Checked)
            {
                numberCondiments=numberCondiments+1;
                txtOrder.Text = txtOrder.Text + "Onions\r\n";
            }
            if (chkPickles.Checked)
            {
                numberCondiments=numberCondiments+1;
                txtOrder.Text = txtOrder.Text + "Pickles\r\n";
            }
            // If no condiments picked, say so
            if (numberCondiments == 0) 
            {
                txtOrder.Text = txtOrder.Text + "No Condiments\r\n";
            }
        }

        private void btnClear_Click(object sender, EventArgs e)
        {
            // Set bread to white
            breadChoice = 1;
            rdoWhite.Checked = true;
            // Clear all meat choices
            chkRoastBeef.Checked = false;
            chkHam.Checked = false;
            chkTurkey.Checked = false;
            chkPastrami.Checked = false;
            chkSalami.Checked = false;
            // Set cheese to none
            cheeseChoice = 0;
            rdoNone.Checked = true;
            // Clear all condiment choices
            chkMustard.Checked = false;
            chkMayo.Checked = false;
            chkLettuce.Checked = false;
            chkTomato.Checked = false;
            chkOnions.Checked = false;
            chkPickles.Checked = false;
            // Clear text box
            txtOrder.Text = "";
        }
    }
}

File: CC_Visual Basic/VisualBasic/BeginVBE/BeginVBE/BVBE Projects/Sandwich/Form1.vb
Public Class Form1
  Dim BreadChoice As Integer
  Dim CheeseChoice As Integer
  Dim NumberMeats As Integer
    Dim NumberCondiments As Integer
    Dim Cost As Single


    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
    'Initialize bread and cheese choices
    BreadChoice = 1
        CheeseChoice = 0
    End Sub

  Private Sub rdoWhite_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles rdoWhite.CheckedChanged
    'White bread selected
    BreadChoice = 1
  End Sub

  Private Sub rdoWheat_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles rdoWheat.CheckedChanged
    'Wheat bread selected
    BreadChoice = 2
  End Sub

  Private Sub rdoRye_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles rdoRye.CheckedChanged
    'Rye bread selected
    BreadChoice = 3
  End Sub

  Private Sub rdoNone_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles rdoNone.CheckedChanged
    'No cheese selected
    CheeseChoice = 0
  End Sub

  Private Sub rdoAmerican_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles rdoAmerican.CheckedChanged
    'American cheese selected
    CheeseChoice = 1
  End Sub

  Private Sub rdoSwiss_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles rdoSwiss.CheckedChanged
    'Swiss cheese selected
    CheeseChoice = 2
  End Sub

  Private Sub btnOrder_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnOrder.Click
    'Start Text with bread type
    txtOrder.Text = "Sandwich Order:" & ControlChars.CrLf
    Select Case BreadChoice
      Case 1
        txtOrder.Text = txtOrder.Text & "White Bread" & ControlChars.CrLf
      Case 2
        txtOrder.Text = txtOrder.Text & "Wheat Bread" & ControlChars.CrLf
      Case 3
        txtOrder.Text = txtOrder.Text & "Rye Bread" & ControlChars.CrLf
    End Select
    'Add and count meats
    NumberMeats = 0
    If chkRoastBeef.Checked Then
      NumberMeats = NumberMeats + 1
      txtOrder.Text = txtOrder.Text & "Roast Beef" & ControlChars.CrLf
    End If
    If chkHam.Checked Then
      NumberMeats = NumberMeats + 1
      txtOrder.Text = txtOrder.Text & "Ham" & ControlChars.CrLf
    End If
    If chkTurkey.Checked Then
      NumberMeats = NumberMeats + 1
      txtOrder.Text = txtOrder.Text & "Turkey" & ControlChars.CrLf
    End If
    If chkPastrami.Checked Then
      NumberMeats = NumberMeats + 1
      txtOrder.Text = txtOrder.Text & "Pastrami" & ControlChars.CrLf
    End If
    If chkSalami.Checked Then
      NumberMeats = NumberMeats + 1
      txtOrder.Text = txtOrder.Text & "Salami" & ControlChars.CrLf
    End If
    'If no meats picked, say so
    If NumberMeats = 0 Then
      txtOrder.Text = txtOrder.Text & "No Meat" & ControlChars.CrLf
    End If
    'Add cheese type
    Select Case CheeseChoice
      Case 0
        txtOrder.Text = txtOrder.Text & "No Cheese" & ControlChars.CrLf
      Case 1
        txtOrder.Text = txtOrder.Text & "American Cheese" & ControlChars.CrLf
      Case 2
        txtOrder.Text = txtOrder.Text & "Swiss Cheese" & ControlChars.CrLf
    End Select
    'Finally, add  and count condiments
    NumberCondiments = 0
    If chkMustard.Checked Then
      NumberCondiments = NumberCondiments + 1
      txtOrder.Text = txtOrder.Text & "Mustard" & ControlChars.CrLf
    End If
    If chkMayo.Checked Then
      NumberCondiments = NumberCondiments + 1
      txtOrder.Text = txtOrder.Text & "Mayonnaise" & ControlChars.CrLf
    End If
    If chkLettuce.Checked Then
      NumberCondiments = NumberCondiments + 1
      txtOrder.Text = txtOrder.Text & "Lettuce" & ControlChars.CrLf
    End If
    If chkTomato.Checked Then
      NumberCondiments = NumberCondiments + 1
      txtOrder.Text = txtOrder.Text & "Tomato" & ControlChars.CrLf
    End If
    If chkOnions.Checked Then
      NumberCondiments = NumberCondiments + 1
      txtOrder.Text = txtOrder.Text & "Onions" & ControlChars.CrLf
    End If
    If chkPickles.Checked Then
      NumberCondiments = NumberCondiments + 1
      txtOrder.Text = txtOrder.Text & "Pickles" & ControlChars.CrLf
    End If
        'If no condiments picked, say so
        If NumberCondiments = 0 Then
            txtOrder.Text = txtOrder.Text & "No Condiments" & ControlChars.CrLf
        End If
        Cost = 3.95
        If NumberMeats < 2 Then
            Cost = Cost + Cost * 0.08
            txtCost.Text = txtCost.Text & Format(Cost, "$0.00")
        Else
            Cost = Cost + (NumberMeats - 1) * 0.75
            Cost = Cost + Cost * 0.08
            txtCost.Text = txtCost.Text & Format(Cost, "$0.00")
        End If
    End Sub

  Private Sub btnClear_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnClear.Click
    'Set bread to white
    BreadChoice = 1
    rdoWhite.Checked = True
    'Clear all meat choices
    chkRoastBeef.Checked = False
    chkHam.Checked = False
    chkTurkey.Checked = False
    chkPastrami.Checked = False
    chkSalami.Checked = False
    'Set cheese to none
    CheeseChoice = 0
    rdoNone.Checked = True
    'Clear all condiment choices
    chkMustard.Checked = False
    chkMayo.Checked = False
    chkLettuce.Checked = False
    chkTomato.Checked = False
    chkOnions.Checked = False
    chkPickles.Checked = False
        'Clear text box
        txtOrder.Text = ""
        txtCost.Text = ""
    End Sub

    Private Sub txtOrder_TextChanged(sender As Object, e As EventArgs) Handles txtOrder.TextChanged

    End Sub
End Class
